aten/src/ATen/native/mkldnn/Normalization.cpp

```diff
@@ -0,0 +1,72 @@
+#include <ATen/ATen.h>
+#include <ATen/Config.h>
+#include <ATen/NativeFunctions.h>
+#include <tuple>
+
+#if !AT_MKLDNN_ENABLED()
+
+namespace at {
+namespace native {
+
+std::tuple<Tensor, Tensor, Tensor> mkldnn_batch_norm(
+    const Tensor& self,
+    const Tensor& weight,
+    const Tensor& bias,
+    const Tensor& running_mean,
+    const Tensor& running_var,
+    bool train,
+    double momentum,
+    double eps) {
+  AT_ERROR("mkldnn_batch_norm: ATen not compiled with MKLDNN support");
+}
+
+} // namespace native
+} // namespace at
+
+#else // AT_MKLDNN_EBABLED
+
+#include <ATen/native/mkldnn/MKLDNNCommon.h>
+
+namespace at {
+namespace native {
+
+std::tuple<Tensor, Tensor, Tensor> mkldnn_batch_norm(
+    const Tensor& input,
+    const Tensor& weight,
+    const Tensor& bias,
+    const Tensor& running_mean,
+    const Tensor& running_var,
+    bool train,
+    double momentum,
+    double eps) {
+  ideep::tensor& x = itensor_from_mkldnn(input);
+  ideep::tensor& w = itensor_from_mkldnn(weight);
+  ideep::tensor& b = itensor_from_mkldnn(bias);
+  ideep::tensor& m = itensor_from_mkldnn(running_mean);
+  ideep::tensor& v = itensor_from_mkldnn(running_var);
+
+  ideep::tensor y;
+
+  if (train) {
```

Comment:
@bddppq , I think there should check whether running status is used for training and infernce, i.e. running_mean and running_var are defined or not.  Another suggestion is that  mkldnn only support batchnorm2d and batchnorm3d, so you shoud make some checks when call mkldnn, perhaps you can see the  [code](https://github.com/XiaobingSuper/pytorch/commit/ed404f4b748a7cf38e9f466743d2d2bc6aedf003)  as reference. thanks!

Reply:
would mkldnn throw with nice message that it's not supported? in this case the input is already mkldnn tensor so it's better to fail if something is not supported

Reply:
@dzhulgakov I'm afraid not. :-(   How about we add an assertion here to guarantee 2d or 3d batchnorm here?